dotnet/src/Functions/Functions.OpenApi/Model/RestApiParameter.cs

```diff
@@ -111,4 +122,27 @@ internal RestApiParameter(
         this.Format = format;
         this.Schema = schema;
     }
+
+    /// <summary>
+    /// Makes the current instance unmodifiable.
+    /// </summary>
+    internal void Freeze()
+    {
+        this._isFrozen = true;
+    }
+
+    /// <summary>
+    /// Throws an <see cref="InvalidOperationException"/> if the <see cref="RestApiParameter"/> is frozen.
+    /// </summary>
+    /// <exception cref="InvalidOperationException"></exception>
+    private void ThrowIfFrozen()
```

Comment:
Based on changes in this PR, it looks like `ThrowIfFrozen` is used only on `string? ArgumentName` property. If the goal is to parse an OpenAPI document and create an instance of `RestApiParameter`/`RestApiPayloadProperty`/`RestApiServerVariable` and set an argument name without a possibility to modify it later, can we use `init` instead of "freeze" logic? In that case, we will catch an attempt to modify this property after initialization in compile-time instead of runtime.

Reply:
The idea is that consumers should be able to modify the `ArgumentName` if they need to after it got initialized by parser. This possibility will be added in the next PR.